src/QsCompiler/Compiler/Templates/QirDriverCpp.tt

```diff
@@ -1,16 +1,342 @@
+    char <#= arg.Name #>,
+        <# } else if (arg.Type == DataType.ArrayType && arg.ArrayType == DataType.PauliType) { #>
+    InteropArray* <#= arg.Name #>,
+        <# } else if (arg.Type == DataType.RangeType) { #>
+    InteropRange* <#= arg.Name #>,
+        <# } else if (arg.Type == DataType.ResultType) { #>
+    char <#= arg.Name #>,
+        <# } else if (arg.Type == DataType.ArrayType && arg.ArrayType == DataType.ResultType) { #>
+    InteropArray* <#= arg.Name #>,
+        <# } else if (arg.Type == DataType.StringType) { #>
+    const char* <#= arg.Name #>,
+        <# } #>
+    <# } #>
+);
+
+const char InteropFalseAsChar = 0x0;
+const char InteropTrueAsChar = 0x1;
+map<string, bool> BoolAsCharMap{
+    {"0", InteropFalseAsChar},
+    {"false", InteropFalseAsChar},
+    {"1", InteropTrueAsChar},
+    {"true", InteropTrueAsChar}};
+
+map<string, PauliId> PauliMap{
+    {"PauliI", PauliId::PauliId_I},
+    {"PauliX", PauliId::PauliId_X},
+    {"PauliY", PauliId::PauliId_Y},
+    {"PauliZ", PauliId::PauliId_Z}};
+
+const char InteropResultZeroAsChar = 0x0;
+const char InteropResultOneAsChar = 0x1;
+map<string, char> ResultAsCharMap{
+    {"0", InteropResultZeroAsChar},
+    {"Zero", InteropResultZeroAsChar},
+    {"1", InteropResultOneAsChar},
+    {"One", InteropResultOneAsChar}
+};
+
+template<typename T>
+unique_ptr<InteropArray> CreateInteropArray(vector<T>& v)
+{
+    unique_ptr<InteropArray> array(new InteropArray(v.size(), v.data()));
+    return array;
+}
+
+unique_ptr<InteropRange> CreateInteropRange(RangeTuple rangeTuple)
+{
+    unique_ptr<InteropRange> range(new InteropRange(rangeTuple));
+    return range;
+}
+
+template<typename T>
+void FreePointerVector(vector<T*>& v)
+{
+    for (auto p : v)
+    {
+        delete p;
+    }
+}
+
+char TranslatePauliToChar(PauliId& pauli)
+{
+    return static_cast<char>(pauli);
+}
+
+template<typename S, typename D>
+void TranslateVector(vector<S>& sourceVector, vector<D>& destinationVector, function<D(S&)> translationFunction)
+{
+    destinationVector.resize(sourceVector.size());
+    transform(sourceVector.begin(), sourceVector.end(), destinationVector.begin(), translationFunction);
+}
+
+InteropRange* TranslateRangeTupleToInteropRangePointer(RangeTuple& rangeTuple)
+{
+    InteropRange* range = new InteropRange(rangeTuple);
+    return range;
+}
+
+const char* TranslateStringToCharBuffer(string& s)
+{
+    return s.c_str();
+}
 
 int main(int argc, char* argv[])
 {
-    return 0;
+    CLI::App app("QIR Standalone Entry Point Inputs Reference");
+
+    // Initialize simulator.
+    unique_ptr<IRuntimeDriver> sim = CreateFullstateSimulator();
+    QirContextScope qirctx(sim.get(), false /*trackAllocatedObjects*/);
+
+    // Add the --simulation-output options.
+    // N.B. This option should be present in all standalone drivers.
+    string simulationOutputFile;
+    CLI::Option* simulationOutputFileOpt = app.add_option(
+        "-s,--simulation-output", simulationOutputFile,
```

Comment:
```suggestion
"--simulation-output", simulationOutputFile,
```

Reply:
Removing the `-s` option to remove the chance of it colliding with an argument named `s`.

Reply:
done